### PR TITLE
Add support for Gitlab CI

### PR DESCRIPTION
**Pull request by wmanley**
Created based on the documentation here: https://docs.gitlab.com/ee/ci/variables/ .  Not tested with real gitlab CI

TODO:

- [x] Test with gitlab CI.
- [x] Find out what test-reporting format it uses and provide that. JUnit XML: https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html#artifactsreportsjunit
- [x] Check and document password/token handling: Specify STBT_AUTH_TOKEN in the GitLab UI, mark it as "masked", then reassign them in your `.gitlab-ci.yml`:

    ```
    variables:
      STBT_AUTH_TOKEN: $STBT_AUTH_TOKEN
    ```

    See https://docs.gitlab.com/ee/ci/variables/

- [x] See what happens when you stop a job: SIGTERM on Unix, SIGKILL on Windows. https://docs.gitlab.com/runner/executors/shell.html#gitlab-131-and-later
- [ ] Document gitlab CI setup

### CODE REVIEW

File: stbt_rig.py [hunk 1]
```diff
@@ -100,6 +100,27 @@ def main(argv):
               For instructions on how to configure your Bamboo job see
               https://stb-tester.com/manual/continuous-integration
 
+            GITLAB CI INTEGRATION:
+
+              We automatically detect if we are running inside a Gitlab CI job.
+              If so, we enable the following behaviours:
+
+              * Read the access token from $CI_BUILD_TOKEN environment variable.
+              * Record various Jenkins parameters as "tags" in the Stb-tester
```
Comment (kalidasya) [inline stbt_rig.py]: typo, Jenkins -> GitlabCI

**Comment by drothlis** [comment]
> TODO: HOW DO RESULTS WORK IN GITLAB CI?

JUnit XML: https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html#artifactsreportsjunit

**Comment by drothlis** [comment]
> TODO: HOW DOES STOPPING JOBS WORK IN GITLAB CI?

SIGTERM on Unix, SIGKILL on Windows:
https://docs.gitlab.com/runner/executors/shell.html#gitlab-131-and-later

File: stbt_rig.py [hunk 2]
```diff
@@ -329,6 +356,25 @@ def cmd_run(args, node):
...
             if value:
                 tags[v] = value
+    elif args.mode == "gitlab":
+        # Record gitlab CI variables as Stb-tester tags.
+        # Gitlab CI changed the variables names in version 9.0 (released
+        # 2017-03-22).  We support both sets of names, but always record the
+        # new name.
+        # See https://docs.gitlab.com/ee/ci/variables/
+        #                    8.x name      9.0+ name
+        for k, old_key in [('CI_JOB_ID', 'CI_BUILD_ID'),
```
Comment (drothlis) [inline stbt_rig.py]: I think the comment is the wrong way round. The current [gitlab docs](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html#predefined-variables-reference) list CI_JOB_* but not CI_BUILD_*.

Comment (drothlis) [reply stbt_rig.py]: Ah yes, the code is correct, just the comment is wrong.

File: stbt_rig.py [hunk 3]
```diff
@@ -400,6 +446,14 @@ def iter_portal_auth_tokens(portal_url, portal_auth_file, mode):
                 "variable bamboo.STBT_AUTH_PASSWORD")
         return
 
+    if mode == "gitlab":
+        token = os.environ.get("CI_BUILD_TOKEN")
```
Comment (drothlis) [inline stbt_rig.py]: GitLab's CI_BUILD_TOKEN (now called CI_JOB_TOKEN) is to access the GitLab API during the job. It isn't to access the Stb-tester Portal.

**Comment by drothlis** [comment]
Rebased onto "main" to fix conflicts. I dropped the compatibility code for GitLab 8 (2017) and addressed the TODO items (just from reading the docs, I don't have a GitLab instance to test against).

**Comment by drothlis** [comment]
Tested by @kalidasya — thanks!

**Pull request merged**